### PR TITLE
test: ensure connection with sharded topology

### PR DESCRIPTION
**Pull request by reggi**
NODE-2827

bringing in test from https://github.com/mongodb/node-mongodb-native/pull/2557

### CODE REVIEW

**Review by mbroadst — APPROVED** [review]

File: test/functional/sharding_connection.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+describe('Sharding (Connection)', function () {
+  before(function () {
+    return setupDatabase(this.configuration);
+  });
+
+  it('Should use sharded topology', {
+    metadata: { requires: { topology: 'sharded' } },
+    test: function () {
+      const client = this.configuration.newClient({});
```
Comment (emadum) [inline test/functional/sharding_connection.test.js:16]: Why was this necessary? If you're not passing custom options into `this.configuration.newClient`, this should be the same as `test: withClient((client, done) => {...})`

**Review by emadum — APPROVED** [review]
LGTM w/ one question about `withClient` usage

File: test/functional/sharding_connection.test.js (hunk 1 above)
Comment (reggi) [reply test/functional/sharding_connection.test.js:16]: Updated! This is a port, and I believe I needed to pass in useUnifiedTopology in 3.6 and that's why I did it this way, but then possibly removed the option here because of the warning?

Comment (reggi) [reply test/functional/sharding_connection.test.js:16]: Damn, i missed this push locally. I think it's OK to leave in, hope we revamp and check through `withClient` calls later.
